Sources/SourceControl/GitRepository.swift

```diff
@@ -323,11 +323,21 @@ public class GitRepository: Repository, WorkingCheckout {
 
     // MARK: Working Checkout Interface
 
+    internal func _areAllBranchesPushed() throws -> Bool {
+        let unpushedBranches = try Process.checkNonZeroExit(
+            args: Git.tool, "-C", path.asString, "log", "--branches", "--not", "--remotes").chomp()
+        return unpushedBranches.isEmpty
+    }
+
+    internal func _isHeadOnRemote() throws -> Bool {
```

Comment:
I added this to validate that `testHasDetatchedUnpushedCommits` is necessary. It's used in the tests to assert that it `hasUnpushedCommits` despite `_areAllBranchesPushed()` being `true`. If you don't think this distinction is worth preserving then I'm happy to inline and remove that assertion in the test.